src/sdk/namespace_fs.js

```diff
@@ -153,7 +153,11 @@ class NamespaceFS {
         return bucket;
     }
 
-    is_server_side_copy(other, params) {
+    async is_server_side_copy(other, params, object_sdk) {
```

Comment:
I can see that you had to change all to async just for this.
But I think we can do differently - in upload_object() we already have handling when link() fails to fallback to _copy_stream() so why not do that check there, where we have easier async flow to use.